grpcgcp/gcp_picker.go

```diff
@@ -59,10 +59,10 @@ func (p *gcpPicker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 	locator := ""
 	var cmd grpc_gcp.AffinityConfig_Command
 
-	if mcfg, ok := p.gb.methodCfg[info.FullMethodName]; ok && hasGcpCtx {
+	if mcfg, ok := p.gb.methodCfg[info.FullMethodName]; ok {
 		locator = mcfg.GetAffinityKey()
 		cmd = mcfg.GetCommand()
-		if cmd == grpc_gcp.AffinityConfig_BOUND || cmd == grpc_gcp.AffinityConfig_UNBIND {
+		if hasGcpCtx && (cmd == grpc_gcp.AffinityConfig_BOUND || cmd == grpc_gcp.AffinityConfig_UNBIND) {
```

Comment:
What's the reason to move this check? Moving this check would read the locator more often, when hasGcpCtx is false.

Reply:
This is intended because now we always need to know method config. We looked up the methodCfg map anyway and looking at a couple fields is inexpensive.